src/Payments/Api.php

```diff
@@ -174,4 +175,58 @@ public function process_legacy_payment( PaymentContext $context, PaymentResult &
 		$result->set_payment_details( array_merge( $result->payment_details, $gateway_result ) );
 		$result->set_redirect_url( $gateway_result['redirect'] );
 	}
+
+	/**
+	 * Verify all dependencies of registered payment methods have been registered.
+	 * If not, remove that payment method script from the list of dependencies
+	 * of Cart and Checkout block scripts so it doesn't break the blocks and show
+	 * an error in the admin.
+	 */
+	public function verify_payment_methods_dependencies() {
+		global $wp_scripts;
+		$payment_method_scripts = $this->payment_method_registry->get_all_active_payment_method_script_dependencies();
+
+		foreach ( $payment_method_scripts as $payment_method_script ) {
+			$deps = $wp_scripts->registered[ $payment_method_script ]->deps;
```

Comment:
Is this array item guaranteed to exist or do you need a guard?

Reply:
Added a guard in 1845b12b57ca5c26a400a8b80413fd655d2e89a0.